ui/component/collectionEdit/view.jsx

```diff
@@ -0,0 +1,437 @@
+    amount,
+    title,
+    description,
+    thumbnailUrl,
+    tags,
+    locations,
+    languages = [],
+    // rest
+    onDone,
+    publishCollectionUpdate,
+    updateError,
+    updatingCollection,
+    publishCollection,
+    creatingCollection,
+    createError,
+    disabled,
+    activeChannelClaim,
+    incognito,
+    collectionId,
+    collection,
+    collectionUrls,
+    collectionClaimIds,
+  } = props;
+  const activeChannelName = activeChannelClaim && activeChannelClaim.name;
+  let prefix = IS_WEB ? `${DOMAIN}/` : 'lbry://';
+  if (activeChannelName && !incognito) {
+    prefix += `${activeChannelName}/`;
+  }
+  const activeChannelId = activeChannelClaim && activeChannelClaim.claim_id;
+  const collectionName = (claim && claim.name) || (collection && collection.name);
+
+  const [nameError, setNameError] = React.useState(undefined);
+  const [bidError, setBidError] = React.useState('');
+  const [params, setParams]: [any, (any) => void] = React.useState(getCollectionParams());
+  const name = params.name;
+  const isNewCollection = !uri;
+  const { replace } = useHistory();
+  const languageParam = params.languages;
+  const primaryLanguage = Array.isArray(languageParam) && languageParam.length && languageParam[0];
+  const secondaryLanguage = Array.isArray(languageParam) && languageParam.length >= 2 && languageParam[1];
+
+  const collectionClaimIdsString = JSON.stringify(collectionClaimIds);
+
+  function parseName(newName) {
+    let INVALID_URI_CHARS = new RegExp(regexInvalidURI, 'gu');
+    return newName.replace(INVALID_URI_CHARS, '-');
+  }
+
+  function setParam(paramObj) {
+    setParams({ ...params, ...paramObj });
+  }
+
+  function getCollectionParams() {
+    const collectionParams: {
+      thumbnail_url?: string,
+      name?: string,
+      description?: string,
+      title?: string,
+      bid: string,
+      languages?: ?Array<string>,
+      locations?: ?Array<string>,
+      tags?: ?Array<{ name: string }>,
+      claim_id?: string,
+      channel_id?: string,
+      claims: ?Array<string>,
+    } = {
+      thumbnail_url: thumbnailUrl,
+      description,
+      bid: String(amount || 0.001),
+      languages: languages || [],
+      locations: locations || [],
+      tags: tags
+        ? tags.map((tag) => {
+            return { name: tag };
+          })
+        : [],
+      claims: collectionClaimIds,
+    };
+
+    if (activeChannelId) {
+      collectionParams['channel_id'] = activeChannelId;
+    }
+
+    if (!claim) {
+      collectionParams['title'] = collectionName;
+      collectionParams['name'] = parseName(collectionName);
+    }
+
+    if (claim) {
+      collectionParams['claim_id'] = claim.claim_id;
+      collectionParams['title'] = title;
+    }
+
+    return collectionParams;
+  }
+
+  React.useEffect(() => {
+    const collectionClaimIds = JSON.parse(collectionClaimIdsString);
+    setParams({ ...params, claims: collectionClaimIds });
+  }, [collectionClaimIdsString, setParams]);
```

Comment:
Should depend on `params` too?  Wondering if this is causing the Anonymous problem.

Reply:
anonymous problem was partly because I wasn't reading 'incognito' prop.